packages/palette/src/elements/Text/Text.tsx

```diff
@@ -0,0 +1,243 @@
+  lineHeight: TextLineHeight
+  letterSpacing?: number
+  fontWeight?: "normal" | "bold"
+}
+
+/**
+ * Names of typographic treatments
+ */
+export const TEXT_TREATMENTS = [
+  "largeTitle",
+  "title",
+  "subtitle",
+  "text",
+  "mediumText",
+  "caption",
+  "small",
+] as const
+
+/**
+ * TextTreatments
+ */
+export type TextTreatments = {
+  [K in typeof TEXT_TREATMENTS[number]]: TextTreatment
+}
+
+/**
+ * Available typographic treatments
+ */
+export const TEXT_VARIANTS: { [key: string]: TextTreatments } = {
+  large: {
+    largeTitle: {
+      fontSize: "size9",
+      lineHeight: "title",
+      letterSpacing: calculateLetterSpacing("size9", "tight"),
+      fontWeight: "normal",
+    },
+    title: {
+      fontSize: "size7",
+      lineHeight: "title",
+      letterSpacing: calculateLetterSpacing("size6", "tight"),
+      fontWeight: "normal",
+    },
+    subtitle: {
+      fontSize: "size6",
+      lineHeight: "title",
+      fontWeight: "normal",
+    },
+    text: {
+      fontSize: "size3",
+      lineHeight: "body",
+      fontWeight: "normal",
+    },
+    mediumText: {
+      fontSize: "size3",
+      lineHeight: "body",
+      fontWeight: "bold",
+    },
+    caption: {
+      fontSize: "size2",
+      lineHeight: "body",
+      fontWeight: "normal",
+    },
+    small: {
+      fontSize: "size1",
+      lineHeight: "body",
+      fontWeight: "normal",
+    },
+  },
+  small: {
+    largeTitle: {
+      fontSize: "size8",
+      lineHeight: "title",
+      letterSpacing: calculateLetterSpacing("size8", "tight"),
+      fontWeight: "normal",
+    },
+    title: {
+      fontSize: "size6",
+      lineHeight: "title",
+      letterSpacing: calculateLetterSpacing("size6", "tight"),
+      fontWeight: "normal",
+    },
+    subtitle: {
+      fontSize: "size5",
+      lineHeight: "title",
+      fontWeight: "normal",
+    },
+    text: {
+      fontSize: "size4",
+      lineHeight: "body",
+      fontWeight: "normal",
+    },
+    mediumText: {
+      fontSize: "size4",
+      lineHeight: "body",
+      fontWeight: "bold",
+    },
+    caption: {
+      fontSize: "size3",
+      lineHeight: "body",
+      fontWeight: "normal",
+    },
+    small: {
+      fontSize: "size2",
+      lineHeight: "body",
+      fontWeight: "normal",
+    },
+  },
+}
+
+/**
+ * Name of typographic treatment
+ */
+export type TextVariant = keyof TextTreatments
+
+/**
+ * Text
+ */
+export type TextProps = TypographyProps &
+  ColorProps & {
+    variant?: TextVariant
+    textColor?: ResponsiveValue<Color>
+  }
+
+const textColor = style({
+  prop: "textColor",
+  cssProperty: "color",
+  key: "colors",
+})
+
+/**
+ * styled functions for Text
+ */
+export const textMixin = compose(
+  typography,
+  color,
+  textColor
+)
+
+/**
+ * Text
+ */
+export const Text = primitives.Text<TextProps>`
+  ${variant({ variants: TEXT_VARIANTS.small })}
+
+  @media (min-width: ${themeGet("breakpoints.0")}) {
+    ${variant({ variants: TEXT_VARIANTS.large })}
+  }
```

Comment:
So, variants supports the responsive value array syntax just fine, but since we're only swapping values at the xs breakpoint I decided to just do this explicitly.

Reply:
How did this end up looking on RN? any impact when encountering `@media`?